source/frontend/src/features/leases/add/AddLeaseContainer.tsx

```diff
@@ -71,6 +71,10 @@ export const AddLeaseContainer: React.FunctionComponent<
   };
 
   const handleSave = () => {
+    if (formikRef?.current?.isValid) {
```

Comment:
note: in situations where the form is a parent of the buttons, you should be able to use the value of the formik isValid directly instead of needing a separate state variable. In this instance you will need the state variable however since the ref will not trigger an update in the SidebarFooter component.